grafana/resource_team_external_group.go

```diff
@@ -0,0 +1,141 @@
+package grafana
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceTeamExternalGroup() *schema.Resource {
+	return &schema.Resource{
+
+		Description: `
+* [Official documentation](https://grafana.com/docs/grafana/latest/enterprise/team-sync/)
+* [HTTP API](https://grafana.com/docs/grafana/latest/http_api/external_group_sync/)
+`,
+
+		CreateContext: CreateTeamExternalGroup,
+		UpdateContext: UpdateTeamExternalGroup,
+		DeleteContext: UpdateTeamExternalGroup,
+		ReadContext:   ReadTeamExternalGroup,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"team_id": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The Team ID",
+			},
+
+			"groups": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "The team external groups list",
+			},
+		},
+	}
+}
+
+func CreateTeamExternalGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	teamID := d.Get("team_id").(int)
+	d.SetId(strconv.FormatInt(int64(teamID), 10))
+	if err := ManageGroups(d, meta); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.Diagnostics{}
+}
+
+func ReadTeamExternalGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client).gapi
+	teamID, _ := strconv.ParseInt(d.Id(), 10, 64)
+	teamGroups, err := client.TeamGroups(teamID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	groupIDs := make([]string, 0, len(teamGroups))
+	for _, teamGroup := range teamGroups {
+		groupIDs = append(groupIDs, teamGroup.GroupID)
+	}
+	d.Set("groups", groupIDs)
+
+	return diag.Diagnostics{}
+}
+
+func UpdateTeamExternalGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if err := ManageGroups(d, meta); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.Diagnostics{}
+}
+
+func ManageGroups(d *schema.ResourceData, meta interface{}) error {
```

Comment:
```suggestion
func manageTeamExternalGroup(d *schema.ResourceData, meta interface{}) error {
```

Sort of a nitpick, but since the entire provider is in the same package there tends to be pretty verbose function names. And they typically include the entire resource or data source name to avoid ambiguity.

Reply:
Another thing to note is that none of the function names in the provider really need to begin with a capital letter as if to suggest they should be exported. It's sort of an inherited convention to only start a function name with a capital letter if it's referenced directly by the resource (`CreateContext`, `DeleteContext`, etc).